### PR TITLE
Fix restart_ray for den launched clusters

### PR DESCRIPTION
**Pull request by mkandler**
Sky status may not be reachable for cluster launched with Den, so fallback to using IPs set directly on the cluster when restarting Ray

### CODE REVIEW

**Comment by mkandler** [comment]
* **#1404** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1404?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mkandler and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/hardware/on_demand_cluster.py [hunk 1]
```diff
@@ -363,9 +363,12 @@ def _start_ray_workers(self, ray_port, env):
         # Find the internal IP corresponding to the public_head_ip and the rest are workers
         internal_head_ip = None
         worker_ips = []
-        stable_internal_external_ips = self._sky_status()[
-            "handle"
-        ].stable_internal_external_ips
+        sky_status = self._sky_status()
```
Comment (carolineechen) [inline runhouse/resources/hardware/on_demand_cluster.py:366]: wondering if we can get rid of all this and just do `for internal, external in self.stable_internal_external_ips` below, since it should already be set by the time `_start_ray_workers` is called in restart server.

not sure if there's any edge case though so fine to keep this for the release and test later on if we can skip this extra sky status/populating

**Review by carolineechen — APPROVED** [review]
wondering if we can get rid of all this and just do `for internal, external in self.stable_internal_external_ips`, since it should already be set by the time _start_ray_workers is called in restart server.

not sure if there's any edge case though so fine to keep this for the release and test later on if we can skip this extra sky status/populating
